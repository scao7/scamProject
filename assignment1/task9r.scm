;task 9 recursive
;task 9 iteration
(define (ramanujanr depth)
	(define (h x)
		(if (< (- x 1.0) depth)
			(* x 
				(sqrt
					(+ (+ x 5.0)
						(h (+ x 1.0))
						)
					)
				)
			0.0
			)
		)
	(h 1.0)
	)
	
(define (main)
	(setPort (open (getElement ScamArgs 1) 'read))
	(define arg (readExpr))
	(println "(ramanujanr " arg ") is " (fmt "%.25f" (ramanujanr arg)))
)
	