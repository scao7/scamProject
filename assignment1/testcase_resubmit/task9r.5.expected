(ramanujanr 20) is 3.9999996200104743770964433
$4$
